src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -2014,4 +2138,594 @@ void CorProfiler::GetAssemblyAndSymbolsBytes(BYTE** pAssemblyArray, int* assembl
+    if (FAILED(hr)) {
+      Warn("Can't load the TypeDef for: ", integration.replacement.target_method.type_name, ", Module: ", module_metadata->assemblyName);
+      continue;
+    }
+
+    // Now we enumerate all methods with the same target method name. (All overloads of the method)
+    auto enumMethods = Enumerator<mdMethodDef>(
+        [metadata_import, integration, typeDef](HCORENUM* ptr, mdMethodDef arr[], ULONG max, ULONG* cnt) -> HRESULT {
+          return metadata_import->EnumMethodsWithName(ptr, typeDef, integration.replacement.target_method.method_name.c_str(), arr, max, cnt);
+        },
+        [metadata_import](HCORENUM ptr) -> void {
+          metadata_import->CloseEnum(ptr);
+        });
+
+    auto enumIterator = enumMethods.begin();
+    while (enumIterator != enumMethods.end()) {
+      auto methodDef = *enumIterator;
+
+      // Extract the function info from the mdMethodDef
+      const auto caller = GetFunctionInfo(module_metadata->metadata_import, methodDef);
+      if (!caller.IsValid()) {
+        Warn("The caller for the methoddef: ", HexStr(&methodDef, sizeof(mdMethodDef)), " is not valid!");
+        enumIterator = ++enumIterator;
+        continue;
+      }
+
+      // We create a new function info into the heap from the caller functionInfo in the stack, to be used later in the ReJIT process
+      auto functionInfo = new FunctionInfo(caller);
+      hr = functionInfo->method_signature.TryParse();
+      if (FAILED(hr)) {
+        Warn("The method signature: ", functionInfo->method_signature.str(), " cannot be parsed.");
+        delete functionInfo;
+        enumIterator = ++enumIterator;
+        continue;
+      }
+
+      // Compare if the current mdMethodDef contains the same number of arguments as the instrumentation target
+      const auto numOfArgs = functionInfo->method_signature.NumberOfArguments();
+      if (numOfArgs != integration.replacement.target_method.signature_types.size() - 1) {
+        Debug("The caller for the methoddef: ", integration.replacement.target_method.method_name, " doesn't have the right number of arguments.");
+        delete functionInfo;
+        enumIterator = ++enumIterator;
+        continue;
+      }
+
+      // Compare each mdMethodDef argument type to the instrumentation target
+      bool argumentsMismatch = false;
+      const auto methodArguments = functionInfo->method_signature.GetMethodArguments();
+      Debug("Comparing signature for method: ", integration.replacement.target_method.type_name, ".", integration.replacement.target_method.method_name);
+      for (unsigned int i = 0; i < numOfArgs; i++) {
+        const auto argumentTypeName = methodArguments[i].GetTypeTokName(metadata_import);
+        const auto integrationArgumentTypeName = integration.replacement.target_method.signature_types[i + 1];
+        Debug("  -> ", argumentTypeName, " = ", integrationArgumentTypeName);
+        if (argumentTypeName != integrationArgumentTypeName) {
```

Comment:
I suggest adding another condition that will not fail argument matching if the parameter typename is `_`, which is currently done in the profiler right now: https://github.com/DataDog/dd-trace-dotnet/blob/master/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp#L936

The only integration method that currently uses this feature is GraphQL because one of the parameter types changed between API versions. Whether or not we should remove this feature is up for debate, but in the meantime let's make sure we're consistent.

```suggestion
        if (argumentTypeName != integrationArgumentTypeName && integrationArgumentTypeName != "_"_W) {
```

Reply:
fixed in https://github.com/DataDog/dd-trace-dotnet/pull/1007/commits/9e7bcb3955022de47db4e80e29c33db9f64b69c2